internal/ethapi/api.go

```diff
@@ -1349,7 +1357,7 @@ func DoEstimateGas(ctx context.Context, b Backend, args TransactionArgs, blockNr
 
 	// We first execute the transaction at the highest allowable gas limit, since if this fails we
 	// can return error immediately.
-	failed, result, err := executeEstimate(ctx, b, args, state.Copy(), header, vanillaGasCap, hi)
+	failed, result, err := executeEstimate(ctx, b, args, state.Copy(), header, vanillaGasCap, hi, &blockNrOrHash)
```

Comment:
Why not update header.Time here instead? Btw, we don't want different executions of the estimate to run against different timestamps, which would be an issue with this implementation.

Reply:
Nice catch! I prioritized updating block context's time over header's but like you said it shouldn't matter for pending blocks. I'll fix this.